2018102147 최인열

## A

In [1]:
import pandas_datareader.data as web
import matplotlib.pyplot as plt
%matplotlib inline

In [9]:
# A1
ibm = web.get_data_yahoo("IBM", start='2016-01-01', end = '2020-12-31')['Adj Close'].pct_change().dropna()
print("최소 수익률 : ", ibm.min() * 100)
print("최대 수익률 : ", ibm.max() * 100)
print("평균 수익률 : ", ibm.mean() * 100)

최소 수익률 :  -12.85070735972511
최대 수익률 :  11.301043618079841
평균 수익률 :  0.023500088350372976


In [68]:
import pandas as pd
ibm_week = []
for i in range(len(ibm)):
    ibm_week.append(ibm[i * 5: (i * 5) + 5].sum())

ibm_week = pd.DataFrame(ibm_week)
ibm_week = ibm_week[ibm_week[0] != 0.000000]

In [55]:
# A2
print("주간 최소 수익률 : ", ibm_week[0].min() * 100)
print("주간 최대 수익률 : ", ibm_week[0].max() * 100)
print("주간 평균 수익률 : ", ibm_week[0].mean() * 100)

주간 최소 수익률 :  -21.56417278003714
주간 최대 수익률 :  12.9715558033769
주간 평균 수익률 :  0.1174071874330142


In [56]:
#A3
print("Annualized mean: " '%.3f%%' % (ibm_week[0].mean() * 252 * 100))

Annualized mean: 29.587%


# B

In [83]:
import pandas as pd
ibm_week = []
for i in range(len(ibm)):
    ibm_week.append(ibm[i * 5: (i * 5) + 5].sum())

ibm_week = pd.DataFrame(ibm_week)
ibm_week = ibm_week[ibm_week[0] != 0.000000]
ibm_week

0
0   -0.044223
1   -0.011743
2   -0.061700
3    0.022646
4    0.027759
..        ...
247 -0.004632
248  0.011646
249  0.004934
250 -0.006653
251  0.009618

[252 rows x 1 columns]

In [84]:
sp500 = web.get_data_yahoo("^GSPC", start='2016-01-01', end='2020-12-31')['Adj Close'].pct_change().dropna()
sp500

Date
2016-01-04   -0.015304
2016-01-05    0.002012
2016-01-06   -0.013115
2016-01-07   -0.023700
2016-01-08   -0.010838
                ...   
2020-12-24    0.003537
2020-12-28    0.008723
2020-12-29   -0.002227
2020-12-30    0.001342
2020-12-31    0.006439
Name: Adj Close, Length: 1259, dtype: float64

In [85]:
sp500_week = []
for i in range(len(sp500)):
    sp500_week.append(sp500[i * 5: (i * 5) + 5].sum())

sp500_week = pd.DataFrame(sp500_week)
sp500_week = sp500_week[sp500_week[0] != 0.000000]
sp500_week

0
0   -0.060946
1   -0.021213
2   -0.001321
3    0.033126
4   -0.044860
..        ...
247  0.010240
248  0.000453
249  0.014827
250 -0.005208
251  0.014276

[252 rows x 1 columns]

In [89]:
ibm_week = pd.Series(ibm_week[0])
sp500_week = pd.Series(sp500_week[0])

In [91]:
import statsmodels.api as sm

Y = ibm_week
X = sm.add_constant(sp500_week)
model = sm.OLS(Y, X)
results = model.fit()
results.params

const   -0.001781
0        1.061367
dtype: float64

In [92]:
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.564
Model:                            OLS   Adj. R-squared:                  0.562
Method:                 Least Squares   F-statistic:                     323.2
Date:                Mon, 17 Oct 2022   Prob (F-statistic):           6.09e-47
Time:                        16:03:16   Log-Likelihood:                 576.16
No. Observations:                 252   AIC:                            -1148.
Df Residuals:                     250   BIC:                            -1141.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         -0.0018      0.002     -1.139      0.256      -0.005       0.001
0              1.0614      0.059     17.976      0.000       0.945       1.178
==============================================================================
Omnibus:                       29.177   Durbin-Watson:                   2.321
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              156.189
Skew:                           0.105   Prob(JB):                     1.21e-34
Kurtosis:                       6.851   Cond. No.                         38.0
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

B1 : 위의 코드에서 1.0614가 시장을 설명하는 beta값임.

B2 : std err로 표기된 0.059 값이 표준 편차

C

In [93]:
ibm_week

0     -0.044223
1     -0.011743
2     -0.061700
3      0.022646
4      0.027759
         ...   
247   -0.004632
248    0.011646
249    0.004934
250   -0.006653
251    0.009618
Name: 0, Length: 252, dtype: float64

In [111]:
ibm_month = []
for i in range(len(ibm_week)):
    ibm_month.append(ibm_week[i * 5: (i * 5) + 5].sum())

ibm_month = pd.DataFrame(ibm_month)
ibm_month = ibm_month[ibm_month[0] != 0.000000]
ibm_month

0
0  -0.067260
1   0.122735
2   0.024660
3   0.048858
4   0.003418
5   0.074958
6  -0.022989
7  -0.021592
8   0.061209
9   0.021672
10  0.057588
11  0.012063
12 -0.036664
13 -0.100194
14  0.021792
15 -0.067867
16 -0.006389
17  0.032553
18  0.027187
19  0.026227
20  0.063208
21 -0.005933
22 -0.006092
23 -0.066833
24 -0.044385
25  0.032772
26  0.031653
27 -0.049988
28 -0.114880
29 -0.116791
30  0.228653
31  0.017219
32  0.067072
33 -0.052110
34  0.025553
35  0.070449
36 -0.033775
37  0.002275
38 -0.037557
39  0.012304
40  0.016801
41 -0.042188
42 -0.021768
43 -0.019077
44  0.043428
45  0.030477
46  0.005434
47  0.008125
48 -0.034677
49  0.071568
50  0.002966

In [105]:
ind5 = web.DataReader('5_industry_Portfolios', 'famafrench', start='2016-01-01', end='2020-12-31')
ind5_ret = ind5[0] / 100

In [106]:
ind5_ret.head()

Cnsmr   Manuf   HiTec   Hlth    Other
Date                                           
2016-01 -0.0329 -0.0338 -0.0484 -0.0997 -0.0829
2016-02  0.0054  0.0135 -0.0055 -0.0095 -0.0014
2016-03  0.0579  0.0803  0.0785  0.0394  0.0705
2016-04 -0.0015  0.0348 -0.0302  0.0494  0.0237
2016-05  0.0072 -0.0009  0.0405  0.0239  0.0090